packages/commonwealth/client/scripts/views/pages/view_thread/ViewThreadPage.tsx

```diff
@@ -153,6 +165,38 @@ const ViewThreadPage = ({ identifier }: ViewThreadPageProps) => {
     );
   }, []);
 
+  useEffect(() => {
+    if (groups && thread?.topic) {
+      const topicId = thread.topic.id;
+      const addressTopicMemberIds = app.user.addresses.map(
+        (ids) => ids.address
+      );
+
+      const groupsWithTopic = groups.filter((community) =>
+        community.topics.some((topic) => topic.id === topicId)
+      );
+
+      // Extract the names of the groups that have the current topic
+      const groupNamesWithTopic = groupsWithTopic.map((community) => ({
+        id: community.id,
+        name: community.name,
+      }));
+
+      // Set the gatedGroups state to the list of group names
+      setGatedGroups(groupNamesWithTopic);
+
+      if (groupsWithTopic.length > 0) {
+        // Check if the user is a member of any of the groups with the current topic
+        const isMemberOfAnyGroup = groupsWithTopic.some((community) =>
+          community.members.some((id) => addressTopicMemberIds.includes(id))
+        );
+
+        // You can use the 'isMemberOfAnyGroup' variable as needed
+        setHideGatingBanner(isMemberOfAnyGroup);
+      }
+    }
+  }, [groups, thread?.topic]);
+
```

Comment:
i would suggest removing this and the `gatedGroups` var from the state (we actually don't need these extra checks)

```diff
// find if the current topic is gated
const foundGatedTopic = groups.find(x => x.topic.find(y => y.id === topicId))

// Check if the user is a member of any of the groups with the current topic
if(foundGatedTopic) {
  const isMemberOfGatedTopic = foundGatedTopic.members.find(x => x?.address?.address === app.user.activeAccount.address)
}
```